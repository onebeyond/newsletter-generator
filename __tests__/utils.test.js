
const { renderNewsletter, getCurrentMonth, getTimeStamp, getDatesRanges } = require('../src/utils')
const fixtures = require('../__fixtures__')
const config = require('../src/config')
jest.useFakeTimers().setSystemTime(new Date('2020-01-01'))

describe('Utils', () => {
  describe('getCurrentMonth', () => {
    it('should return January', () => {
      expect(getCurrentMonth()).toBe('January')
    })
  })
  describe('getDatesRanges', () => {
    it('should return a valid range for 90 days', () => {
      const { start, end } = getDatesRanges(90)
      expect(end).toBe('2020-01-01')
      expect(start).toBe('2019-10-03')
    })
    it('should return a valid range for 30 days', () => {
      const { start, end } = getDatesRanges(30)
      expect(end).toBe('2020-01-01')
      expect(start).toBe('2019-12-02')
    })
  })

  describe('getTimeStamp', () => {
    it('should return a valid timestamp', () => {
      expect(getTimeStamp()).toBe(1577836800)
    })
  })

  describe('renderNewsletter', () => {
    it('should render a newsletter', async () => {
      const data = {
        currentMonth: 'January',
        stepSecurity: {
          top10: fixtures.stepSecurity.repositories.output.top10,
          average: fixtures.stepSecurity.repositories.output.average
        },
        cauldron: fixtures.cauldron.output,
        npmStat: fixtures.npmStat.output,
        github: fixtures.github.output
      }
      const newsletter = await renderNewsletter({ data, config })
      expect(newsletter).toMatchSnapshot()
    })
  })
})
