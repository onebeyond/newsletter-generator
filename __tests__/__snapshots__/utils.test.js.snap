// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils renderNewsletter should render a newsletter 1`] = `
"# January's Newsletter

## üì¢ Announcements and Updates

<!--- @TODO: Manual addition --->

## üìö Great Resources

<!--- @TODO: Manual addition --->

## üóì Last Meetings

<!--- @TODO: Manual addition --->

## üëÄ Repo of the month

<!--- @TODO: Manual addition --->

## üçø New Repositories

- [onebeyond/newsletter-generator](https://github.com/onebeyond/newsletter-generator)
- [onebeyond/shieldjs](https://github.com/onebeyond/shieldjs)

## üì¶ Latest Releases

- [onebeyond/onebeyond-studio-file-storage@v7.0.0.2](https://github.com/onebeyond/onebeyond-studio-file-storage/releases/tag/v7.0.0.2)
- [onebeyond/swagger-endpoint-validator@v4.0.0](https://github.com/onebeyond/swagger-endpoint-validator/releases/tag/v4.0.0)
- [onebeyond/systemic-knex@ob-v4.0.0](https://github.com/onebeyond/systemic-knex/releases/tag/ob-v4.0.0)

## ü•á Rankings

### Top 10 Stars

| Repository | Total Stars | 
| -- | -- |
| [onebeyond/rascal](https://github.com/onebeyond/rascal) | 400 ‚≠êÔ∏è |
| [guidesmiths/cybersecurity-handbook](https://github.com/guidesmiths/cybersecurity-handbook) | 104 ‚≠êÔ∏è |
| [onebeyond/systemic](https://github.com/onebeyond/systemic) | 54 ‚≠êÔ∏è |
| [guidesmiths/marv](https://github.com/guidesmiths/marv) | 43 ‚≠êÔ∏è |
| [onebeyond/react-form-builder](https://github.com/onebeyond/react-form-builder) | 29 ‚≠êÔ∏è |
| [onebeyond/monaco](https://github.com/onebeyond/monaco) | 25 ‚≠êÔ∏è |
| [onebeyond/generator-systemic](https://github.com/onebeyond/generator-systemic) | 24 ‚≠êÔ∏è |
| [guidesmiths/whoosh](https://github.com/guidesmiths/whoosh) | 14 ‚≠êÔ∏è |
| [onebeyond/confabulous](https://github.com/onebeyond/confabulous) | 13 ‚≠êÔ∏è |
| [onebeyond/interview-code-challenges](https://github.com/onebeyond/interview-code-challenges) | 12 ‚≠êÔ∏è |


### Top 10 Downloads

Total 5908504 downloads for the 80 npm packages since their publication

| Package | Total Downloads | 
| -- | -- |
| [rascal](https://www.npmjs.com/package/rascal) | 2181445 |
| [stashback](https://www.npmjs.com/package/stashback) | 1955026 |
| [@guidesmiths/generator-react-component](https://www.npmjs.com/package/@guidesmiths/generator-react-component) | 437497 |
| [react-native-uservoice](https://www.npmjs.com/package/react-native-uservoice) | 314876 |
| [marv](https://www.npmjs.com/package/marv) | 307220 |
| [marv-pg-driver](https://www.npmjs.com/package/marv-pg-driver) | 231643 |
| [make-manifest](https://www.npmjs.com/package/make-manifest) | 145833 |
| [@onebeyond/license-checker](https://www.npmjs.com/package/@onebeyond/license-checker) | 114225 |
| [whoosh](https://www.npmjs.com/package/whoosh) | 111288 |
| [systemic](https://www.npmjs.com/package/systemic) | 109381 |

_Powered by [Npm Stat](https://github.com/pvorb/npm-stat.com)_


## üìä Health report

The full reports are available at [Cauldron](https://cauldron.io/project/7192) and [Grafana Dashboard](https://cauldron.io/project/7192/public-kibana).

### Activity

| | Commits | Issues |  Reviews | 
| -- | -- | -- | -- |
| Total | 3703 | 442 | 984 |
| Total (Last year)	| 946 | 155 | 390 |
| Year-over-year | 84.05% | 106.67% | 61.16 |

<details>
    <summary>Details</summary>

- Commits: 407
- Lines/commit: 1132.84
- Lines/commit/file: 0.22
- Issues created: 69
- Issues closed: 41
- Issues open: 134
- Pull requests created: 185
- Pull requests closed: 154
- Pull requests open: 190

</details>


### Community

| | Commit authors | Issue submitters |  Review submitters | 
| -- | -- | -- | -- |
| Total | 158 | 129 | 93 |
| Total (Last year)	| 56 | 26 | 34 |
| Year-over-year | 19.15% | -36.59% | 17.24 |

<details>
    <summary>Details</summary>

**Active People**
- Git Authors: 29
- Issue Authors: 10
- PRs Submitters: 20

**Onboarding**
- Git Authors: 15
- Issue Authors: 5
- PRs Submitters: 10

</details>


### Performance


| | Issues | Reviews | 
| -- | -- | -- |
| Total | 6.47 days | 0.85 days |
| Total (Last year)	| 13.43 days | 0.63 days |
| Year-over-year | 126.09% | -89.95% |


<details>
    <summary>Details</summary>

**Issues**
- Time Open (average): 442.48 days
- Time Open (median): 165.69 days
- Open Issues: 135

**Reviews**
- Time Open (average): 419.86 days
- Time Open (median): 382.67 days
- Open Reviews: 202

</details>


## üîê Security

### OpenSSF Scorecard

The average score is 4.34/10

The full reports are available at [Step Security Dashboard (Current)](https://app.stepsecurity.io/github/app/onebeyond/onebeyond/repositories) and [Step Security Dashboard (Legacy)](https://app.stepsecurity.io/github/app/onebeyond/guidesmiths/repositories).


**Top 10 Projects in the OpenSSF Scorecard**


| Repository | Score |
| -- | -- |
| [onebeyond/systemic-knex](https://github.com/onebeyond/systemic-knex) | 6.9 |
| [onebeyond/onebeyond-studio-core](https://github.com/onebeyond/onebeyond-studio-core) | 6.8 |
| [onebeyond/onebeyond-studio-email-providers](https://github.com/onebeyond/onebeyond-studio-email-providers) | 6.8 |
| [onebeyond/swagger-endpoint-validator](https://github.com/onebeyond/swagger-endpoint-validator) | 6.8 |
| [onebeyond/onebeyond-studio-obelisk](https://github.com/onebeyond/onebeyond-studio-obelisk) | 6.5 |
| [onebeyond/cuckoojs](https://github.com/onebeyond/cuckoojs) | 6.3 |
| [onebeyond/open-source-project-template](https://github.com/onebeyond/open-source-project-template) | 6.3 |
| [onebeyond/.github](https://github.com/onebeyond/.github) | 6.2 |
| [onebeyond/12-factor-app-pills](https://github.com/onebeyond/12-factor-app-pills) | 6.2 |
| [onebeyond/systemic-pg](https://github.com/onebeyond/systemic-pg) | 6.2 |

_If you want to report any issue or improve this Newsletter, Checkout [this repository](https://github.com/onebeyond/newsletter-generator)_"
`;
